src/EnergyPlus/AirflowNetwork/src/Solver.cpp

```diff
@@ -12128,6 +12191,726 @@ namespace AirflowNetwork {
+
+        int NodeLoopSupply = 0;
+        int NodeLoopReturn = 0;
+        int NodeSplitter = 0;
+        int NodeMixer = 0;
+        int NodeZoneIntlet = 0;
+        int NodeZoneReturn = 0;
+        int NumOfBranches = 0;
+        int AFNNodeNum;
+        int AFNLinkNum;
+        int AFNLinkNum1;
+        Real64 SumLength = 0.0;
+        Real64 DynamicLoss = 0.0;
+        Real64 MaxRough = 0.0;
+        bool DuctSizingSTFlag;
+        bool DuctSizingSBFlag;
+        bool DuctSizingRTFlag;
+        bool DuctSizingRBFlag;
+        Real64 hydraulicDiameter = 0.0;
+        Array1D<Real64> Par(9); // Parameters passed to RegulaFalsi
+        Real64 SupplyTrunkD = 0.0;
+        Real64 SupplyTrunkArea = 0.0;
+        Real64 SupplyBranchD = 0.0;
+        Real64 SupplyBranchArea = 0.0;
+        Real64 ReturnTrunkD = 0.0;
+        Real64 ReturnTrunkArea = 0.0;
+        Real64 ReturnBranchD = 0.0;
+        Real64 ReturnBranchArea = 0.0;
+        Real64 MdotBranch = 0.0;
+        int SolFla = 0;
+
+        int ZoneNum;
+        int NumOfCtrlZones = 0;
+        for (ZoneNum = 1; ZoneNum <= m_state.dataGlobal->NumOfZones; ++ZoneNum) {
+            if (!m_state.dataZoneEquip->ZoneEquipConfig(ZoneNum).IsControlled) continue;
+            NumOfCtrlZones++;
+            for (int EquipTypeNum = 1; EquipTypeNum <= m_state.dataZoneEquip->ZoneEquipList(ZoneNum).NumOfEquipTypes; ++EquipTypeNum) {
+                if (m_state.dataZoneEquip->ZoneEquipList(ZoneNum).EquipTypeEnum(EquipTypeNum) == DataZoneEquipment::ZoneEquip::AirDistUnit) {
+                    int AirDistUnitNum = m_state.dataZoneEquip->ZoneEquipList(ZoneNum).EquipIndex(EquipTypeNum);
+                    MdotBranch = m_state.dataDefineEquipment->AirDistUnit(AirDistUnitNum).MassFlowRateTU;
+                    break;
+                }
+            }
+        }
+        if (NumOfCtrlZones != 1) {
+            ShowWarningError(m_state, "AirflowNetwork Duct Sizing: The current restriction is limited to a single controlled zone only");
+            ShowContinueError(m_state, format("The number of controlled zone is {}", NumOfCtrlZones));
+            ShowContinueError(m_state, "..Duct sizing is not performed");
+            AirflowNetworkSimu.AFNDuctAutoSize = false;
+            AirflowNetworkSimu.iWPCCnt = iWPCCntr::Input;
+            AirflowNetworkSimu.AllowSupportZoneEqp = false;
+        }
+        Real64 factor = AirflowNetworkSimu.DuctSizeFactor;
+
+        NodeLoopSupply = m_state.dataAirLoop->AirToZoneNodeInfo(1).ZoneEquipSupplyNodeNum(1);
+        NodeLoopReturn = m_state.dataAirLoop->AirToZoneNodeInfo(1).ZoneEquipReturnNodeNum(1);
+        for (AFNNodeNum = 1; AFNNodeNum <= AirflowNetworkNumOfNodes; AFNNodeNum++) {
+            if (AirflowNetworkNodeData(AFNNodeNum).EPlusTypeNum == iEPlusNodeType::SPL) {
+                NodeSplitter = AFNNodeNum;
+            }
+            if (AirflowNetworkNodeData(AFNNodeNum).EPlusTypeNum == iEPlusNodeType::MIX) {
```

Comment:
Some of these can be `else if` rather than `if`, which is faster. Some of them can even be `switch/case`.